internal/services/contracts_events_consumer.go

```diff
@@ -545,14 +551,16 @@ func (c *ContractsEventsConsumer) handleTokensTransferredForDevice(ctx context.C
 
 	if common.HexToAddress(c.settings.AftermarketDeviceAddr) == args.DeviceNftProxy {
 		reward.AftermarketTokenID = null.IntFrom(int(args.DeviceNode.Int64()))
-		reward.AftermarketEarnings = null.IntFrom(int(args.Amount.Int64()))
+		reward.AftermarketEarnings = types.NewNullDecimal(decimal.New(args.Amount.Int64(), 0))
```

Comment:
Converting to `int64` is going to be disastrous, I think. You don't want to lose any precision here.

Reply:
I would have a test where the number is so big that it doesn't fit in `int64`.

Reply:
The decimal.new function expects a int64.

Reply:
Then don't use that function! We have lots of examples of using these types in devices-api and rewards-api—you've experienced it yourself with the privileges work, with the event processing work. See here, for example

https://github.com/DIMO-Network/devices-api/blob/d176e6fc86ac8f69de6688073d007585e44d55e0/internal/services/contracts_events_consumer.go#L214

Reply:
You can also look at https://pkg.go.dev/github.com/ericlagergren/decimal and search for big.Int

Reply:
> I would have a test where the number is so big that it doesn't fit in `int64`.

Did we do this?